### PR TITLE
Implementation of BACE

### PR DESCRIPTION
**Pull request by brookehus**
- [x] Implement feature / fix bug
 - [x] Add tests
 - [x] Update changelog

This moves BACE over to MSMB3 (finally!). I prioritized fidelity to MSMB2 results. I removed the option for sparse matrices that MSMB2 had. The distance matrix has its reciprocal taken twice for the old sparse matrix formulation, and I didn't remove this, so there will be some precision issues. However, these precision issues were present in the original version of BACE so it will return the same result.

The code is as much copied and pasted from Greg's code as possible.

### CODE REVIEW

**Comment by brookehus** [comment]
cc @nhstanley

**Comment by nhstanley** [comment]
Nice! Will allow us to close #655

File: msmbuilder/lumping/bace.py [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+                i = 0
+                while dest.shape[0] > i:
+                    if i + self.chunk_size > dest.shape[0]:
+                        indices.append((s, dest[i:]))
+                    else:
+                        indices.append((s, dest[i:i + self.chunk_size]))
+                    i += self.chunk_size
+        return indices
+
+    def mergeTwoClosestStates(self, c, w, indRecalc, dMat,
```
Comment (cxhernandez) [inline msmbuilder/lumping/bace.py]: Should class functions such as this one be hidden from users (e.g. `_mergeTwoClosestStates`)?

Comment (brookehus) [reply msmbuilder/lumping/bace.py]: yeah that's a good idea I'll hide them all

Comment (cxhernandez) [reply msmbuilder/lumping/bace.py]: 👍 LGTM otherwise

**Pull request merged**